chpl/chpl-service/src/test/java/gov/healthit/chpl/manager/auth/UserManagerTest.java

```diff
@@ -0,0 +1,450 @@
+                .deleteAcl(ArgumentMatchers.any(ObjectIdentity.class), ArgumentMatchers.anyBoolean());
+
+        Mockito.when(bCryptPasswordEncoder.encode(ArgumentMatchers.anyString()))
+                .thenAnswer(i -> i.getArgument(0));
+
+        Mockito.when(errorMessageUtil.getMessage(ArgumentMatchers.anyString()))
+                .thenAnswer(i -> i.getArgument(0));
+
+        Mockito.when(env.getProperty("authMaximumLoginAttempts"))
+                .thenReturn(MAX_FAILED_LOGIN_ATTEMPTS);
+        Mockito.when(env.getProperty("resetLinkExpirationTimeInHours"))
+                .thenReturn(RESET_LINK_EXPIRATION);
+
+        Mockito.doNothing()
+                .when(activityManager)
+                .addActivity(ArgumentMatchers.any(), ArgumentMatchers.anyLong(), ArgumentMatchers.anyString(),
+                        ArgumentMatchers.any(), ArgumentMatchers.any());
+
+        Mockito.doNothing()
+                .when(userResetTokenDAO)
+                .deletePreviousUserTokens(ArgumentMatchers.anyLong());
+        Mockito.when(userResetTokenDAO.create(ArgumentMatchers.anyString(), ArgumentMatchers.anyLong()))
+                .thenReturn(UserResetTokenDTO.builder().build());
+        Mockito.when(userResetTokenDAO.findByAuthToken(ArgumentMatchers.anyString()))
+                .thenReturn(UserResetTokenDTO.builder()
+                        .creationDate(new Date())
+                        .build());
+    }
+
+    @Test
+    public void create_GoodData_Success() throws UserCreationException {
+
+        UserDTO user = getUserDTO(1L);
+        UserManager userManager = new UserManager(null, userDAO, null, bCryptPasswordEncoder, mutableAclService, null, null);
+
+        UserDTO newUser = userManager.create(user, "@ThisShouldBeAStrongPassword1!");
+
+        assertNotNull(newUser);
+    }
+
+    @Test(expected = UserCreationException.class)
+    public void create_WeakPassword_UserCreationException() throws UserCreationException {
+        Mockito.when(userDAO.create(ArgumentMatchers.any(UserDTO.class), ArgumentMatchers.anyString()))
+                .thenThrow(new UserCreationException());
+
+        UserDTO user = getUserDTO(1L);
+
+        UserManager userManager = new UserManager(null, userDAO, null, bCryptPasswordEncoder, mutableAclService, null, null);
+
+        userManager.create(user, "@ThisShouldBeAStrongPassword1!");
+
+        fail();
+    }
+
+    @Test
+    public void update_GoodData_SuccessAndActivityWritten()
+            throws UserRetrievalException, JsonProcessingException, EntityCreationException, EntityRetrievalException,
+            ValidationException {
+
+        UserManager userManager = new UserManager(null, userDAO, null, null, null, null, activityManager);
+
+        UserDTO user = getUserDTO(1L);
+        UserDTO updatedUser = userManager.update(user);
+
+        assertNotNull(updatedUser);
+
+        Mockito.verify(activityManager).addActivity(ArgumentMatchers.any(), ArgumentMatchers.anyLong(),
```

Comment:
I probably just don't understand how this works. How does this work? What is being verified? Also, the method name says "SuccessAndActivityWritten". I'm assuming the "ActivityWritten" is this part. Is the "Success" part expected to be verified too?